src/librustc/ty/print/pretty.rs

```diff
@@ -891,21 +891,25 @@ pub trait PrettyPrinter<'tcx>:
 
         match (ct.val, &ct.ty.kind) {
             (_,  ty::FnDef(did, substs)) => p!(print_value_path(*did, substs)),
-            (ty::ConstKind::Unevaluated(did, substs), _) => {
-                match self.tcx().def_kind(did) {
-                    | Some(DefKind::Static)
-                    | Some(DefKind::Const)
-                    | Some(DefKind::AssocConst) => p!(print_value_path(did, substs)),
-                    _ => if did.is_local() {
-                        let span = self.tcx().def_span(did);
-                        if let Ok(snip) = self.tcx().sess.source_map().span_to_snippet(span) {
-                            p!(write("{}", snip))
+            (ty::ConstKind::Unevaluated(did, substs, promoted), _) => {
+                if let Some(promoted) = promoted {
+                    p!(write("{:?}", promoted));
```

Comment:
keep printing `p!(print_value_path(did, substs))` before the promoted id, it may come from another function after inlining

Reply:
Done!